robbery.js

```diff
@@ -6,6 +6,184 @@
+    var end = deadline;
+    var busyTime = Object.keys(gang)
+        .reduce(concatSchedule(gang), [])
+        .sort(compareTime);
+    var freeTime = [];
+    var finish = busyTime.reduce(concatFreeTime(freeTime, duration), 0);
+    if (finish + duration <= end) {
+        freeTime.push({ from: finish, to: end });
+    }
+
+    return freeTime;
+}
+
+function robberSchedule(schedule, bankTimeZone) {
+    return function (acc, robber) {
+        acc[robber] = schedule[robber].map(getFormatSchedule(bankTimeZone));
+
+        return acc;
+    };
+}
+
+function freeGangSchedule(schedule, bankTimeZone, duration) {
+    var formatedSchedule = Object.keys(schedule).reduce(robberSchedule(schedule, bankTimeZone), {});
+
+    return createGangFreeTime(formatedSchedule, duration);
+}
+
+function compareTime(date1, date2) {
+    return date1.from - date2.from;
+}
+
+function concatSchedule(gang) {
+    return function (acc, robber) {
+        return acc.concat(gang[robber]);
+    };
+}
+
+function getBankScheduleForWeek(workingHours) {
+    return function (weekDay) {
+        var dayFrom = weekDay + ' ' + workingHours.from;
+        var dayTo = weekDay + ' ' + workingHours.to;
+
+        return { from: dayFrom, to: dayTo };
+    };
+}
+
+function formatBankSchedule(workingHours, timeZone) {
+    var weekWorkingHours = daysArray.map(getBankScheduleForWeek(workingHours));
+
+    return weekWorkingHours.map(getFormatSchedule(timeZone));
+}
+
+function findFirstMoment(sortedSchedule, duration, startTime) {
+    var endGrub = startTime;
+
+    for (var i = 0; i < sortedSchedule.length; i++) {
+        var start = sortedSchedule[i].from;
+        var end = sortedSchedule[i].to;
+        var finish = end < endGrub ? end : endGrub;
+        if (start + duration <= finish) {
+            return start;
+        }
+        endGrub = end > endGrub ? end : endGrub;
+    }
+
+    return false;
+}
+
+function filterByStartTime(startTime) {
+    return function (time) {
+        if (time.from < startTime && time.to > startTime) {
+            time.from = startTime;
+        }
+
+        return time.from >= startTime;
+    };
+}
+
+function getAppropriateSchedule(schedule, duration, workingHours) {
+    var bankTimeZone = parseInt(TIME_REG.exec(workingHours.from)[5]);
+    var freeGangTime = freeGangSchedule(schedule, bankTimeZone, 0, duration);
+
+    var bankSchedule = formatBankSchedule(workingHours, bankTimeZone);
+
+    return bankSchedule.concat(freeGangTime);
+}
+
+function getTheMoment(schedule, startTime, duration) {
+    var sortedSchedule = schedule
+        .filter(filterByStartTime(startTime))
+        .sort(compareTime);
+
+    return findFirstMoment(sortedSchedule, duration, startTime);
+}
+
+function getCountOfDuration(timeInMinute, duration) {
+    var count = 0;
+    while (Math.floor(timeInMinute / duration) !== 0) {
+        count++;
+        timeInMinute -= duration;
+    }
+
+    return count;
+}
+
+function padZeros(number) {
```

Comment:
`padZeros`?
![Image of Sub-Zero](http://vignette1.wikia.nocookie.net/mortalkombat/images/7/7d/Rev1_Cl-Sub-Zero_versus.png/revision/latest?cb=20120811183330&path-prefix=ru)